src/v/security/audit/schemas/types.h

```diff
@@ -176,11 +184,33 @@ struct actor {
     auto equality_fields() const { return std::tie(authorizations, user); }
 };
 
+struct acl_binding_detail {
+    std::optional<ss::sstring> resource_type;
+    std::optional<ss::sstring> resource_name;
+    std::optional<ss::sstring> pattern_type;
+    std::optional<ss::sstring> acl_principal;
+    std::optional<ss::sstring> acl_host;
+    std::optional<ss::sstring> acl_operation;
+    std::optional<ss::sstring> acl_permission;
+
+    auto equality_fields() const {
+        return std::tie(
+          resource_type,
+          resource_name,
+          pattern_type,
+          acl_principal,
+          acl_host,
+          acl_operation,
+          acl_permission);
+    }
+};
+
 struct resource_detail {
     ss::sstring name;
     ss::sstring type;
+    std::optional<std::variant<acl_binding_detail>> data;
```

Comment:
why have a variant of one type?

Reply:
Had that in there in case I found other resources I wanted to audit in addition to `acl_binding_detail`.  I can remove it

Reply:
nit: this is still resolved